### PR TITLE
Extend API coverage to all endpoints

### PR DESCRIPTION
**Pull request by mandoonandy**
Not a merge request - wait for testing

Something to look out.
Testing to be discussed.
Still requires validation against working installation.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -124,10 +115,7 @@ func (c *Client) CreateNetwork(netblock string, provideDHCP bool, provideNAT boo
 func (c *Client) DeleteNetwork(networkUUID string) error {
 	path := fmt.Sprintf("networks/%s", networkUUID)
 	_, err := c.httpRequest(path, "DELETE", bytes.Buffer{})
-	if err != nil {
-		return err
-	}
-	return nil
+	return err
```
Comment (mikalstill) [inline client.go:93]: LOL, good call.

File: client.go [hunk 2]
```diff
@@ -265,37 +253,192 @@ func (c *Client) CreateInstance(Name string, CPUs int, Memory int,
...
+	_, err = c.httpRequest(path, "POST", *bytes.NewBuffer(post))
+
+	return err
+}
+
+// Snapshot defines a snapshot of an instance
+type Snapshot struct {
+	UUID    string `json:"uuid"`
+	Device  string `json:"device"`
+	Created string `json:"created"` // TODO: Change to time.Time
```
Comment (mikalstill) [inline client.go]: The result is a float Unix epoch time IIRC.

File: client.go [hunk 3]
```diff
@@ -265,37 +253,192 @@ func (c *Client) CreateInstance(Name string, CPUs int, Memory int,
...
+	_, err := c.httpRequest(path, "POST", bytes.Buffer{})
+	return err
+}
+
+// UnPauseInstance will unpause an instance
+func (c Client) UnPauseInstance(instanceUUID string) error {
+	path := fmt.Sprintf("instances/%s/unpause", instanceUUID)
+	_, err := c.httpRequest(path, "POST", bytes.Buffer{})
+	return err
+}
```
Comment (mikalstill) [inline client.go:263]: Do you think that all this repetitive code should be done more like:

```
func (c *Client) thingInstance(instanceUUID string, thing string) error {
    path := fmt.Sprintf("instances/%s/%s", instanceUUID, thing)
    _, err := c.httpRequest(path, "POST", bytesBuffer{})
    return err
}

func (c *Client) UnPauseInstance(instanceUUID string) error {
    return thingInstance(instanceUUID, "unpause")
}
```

If so, is there a nice way we could handle the case where its not the URL which changes, but the HTTP verb instead?

**Review by mikalstill — COMMENTED** [review]
Thanks for this.

File: client.go (hunk 2 above)
Comment (mandoonandy) [reply client.go]: Endpoint /nodes appears to return an RFC822 string in "lastseen".
Endpoint /instances/uuid/events returns a timestamp as a float?
My preference is for golang to convert the value to time.Time.

Comment (mikalstill) [reply client.go]: Oh, nice catch on the inconsistency. So you're saying I should always just return epoch seconds?

Comment (mandoonandy) [reply client.go]: maybe milliseconds?

Comment (mandoonandy) [reply client.go]: ignore me. stick with seconds.

Comment (mikalstill) [reply client.go]: Python likes epoch seconds as a float, so let's go with that. I've filed shakenfist/shakenfist#70 to track that.

**Comment by mandoonandy** [comment]
This is ready for review.
Not fully tested against s8f since some endpoints have unexplained errors. eg. example.go fails at "Get a specific network"

**Comment by mikalstill** [comment]
LGTM, let's dig into the API errors tomorrow, it might also be your test cluster.

**Pull request merged**